### PR TITLE
docs: Add CatalystDataGatewayRepository docs

### LINKED ISSUE #350: 🛠️ [TASK] : Update documentation and provide usage examples
### Summary

Update documentation and provide usage examples

### PR DESCRIPTION
**Pull request by coire1**
# Description

This PR adds usage examples for the `CatalystDataGatewayRepository` and fixes the failing tests.

## Related Issue(s)

List the issue numbers related to this pull request.

Closes #350

### CODE REVIEW

**Review by stevenj — CHANGES_REQUESTED** [review]
Rather than as individual readme files, I think we should have this kind of documentation in the docs themselves.  Can they be moved to the docs in a section for our Dart API's?

**Comment by coire1** [comment]
@stevenj we agreed with @minikin to movethe content of the README as code comments, leaving the README as it was before. 
Mock instructions are not necessary anymore.

File: catalyst_voices/packages/catalyst_voices_repositories/test/src/catalyst_data_gateway_repository/catalyst_data_gateway_repository_test.dart [hunk 1]
```diff
@@ -1,82 +1,136 @@
...
+// implementations outside of its own library". For this reason
+// `@GenerateNiceMocks` doesn't work as intended and we opted to
+// mock the `CatGatewayApi` using `Fake`.
+class FakeCatGatewayApi<T> extends Fake implements CatGatewayApi {
+  final chopper.Response<T> response;
 
+  FakeCatGatewayApi(this.response);
+
+  @override
+  Future<chopper.Response<dynamic>> apiHealthStartedGet() async {
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_repositories/test/src/catalyst_data_gateway_repository/catalyst_data_gateway_repository_test.dart:21]: How about using expressions here?

```
@override
Future<chopper.Response<dynamic>> apiHealthStartedGet() async => response;
```

For one-liners, especially if there are plenty of them helps to reduce the vertical space it takes

File: catalyst_voices/packages/catalyst_voices_repositories/test/src/catalyst_data_gateway_repository/catalyst_data_gateway_repository_test.dart [hunk 2]
```diff
@@ -1,82 +1,136 @@
...
+      final result = await repository.getHealthReady();
       expect(result.isFailure, true);
       expect(result.failure, equals(NetworkErrors.serviceUnavailable));
     });
 
     test('getHealthLive success', () async {
-      when(mock.getHealthLive()).thenAnswer((_) async => Success(null));
-      final result = await mock.getHealthLive();
+      final repository = setupRepository(
+        chopper.Response(http.Response('', 200), null),
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_repositories/test/src/catalyst_data_gateway_repository/catalyst_data_gateway_repository_test.dart:116]: Would `HttpStatus.ok` status work here? It gives a meaningful name to a status code. Wdyt? There will be more places to replace this if you agree:)

**Comment by stevenj** [comment]
As this is not a M2 deliverable, can we retarget the PR to the `feat/m3` branch?
I am trying to reduce churn on `main` leading up to tagging it as the Milestone 2 release.

**Pull request merged**